File: Typescript/BMICalculator.ts

class BMICalculator {
    calculateBMI(weight: number, height: number): number {
      
        const heightInMeters = height / 100; 
        const bmi = weight / (heightInMeters * heightInMeters);
        return parseFloat(bmi.toFixed(2));
    }

    getCategory(bmi: number): string {
        if (bmi < 18.5) {
            return 'Underweight';
        } else if (bmi >= 18.5 && bmi < 24.9) {
            return 'Normal weight';
        } else if (bmi >= 25 && bmi < 29.9) {
            return 'Overweight';
        } else {
            return 'Obese';
        }
    }
}

const bmiCalculator = new BMICalculator();

const weight = 70; 
const height = 175; 

const bmiResult = bmiCalculator.calculateBMI(weight, height);
const bmiCategory = bmiCalculator.getCategory(bmiResult);

console.log(`BMI: ${bmiResult}`);
console.log(`Category: ${bmiCategory}`);

File: C++/BMICalculator.cpp
#include <iostream>
#include <cmath>

using namespace std;

// Function to calculate BMI
double calculateBMI(double weight, double height) {
    return weight / pow(height, 2);
}

int main() {
    // Input weight in kilograms and height in meters
    double weight, height;

    cout << "Enter weight in kilograms: ";
    cin >> weight;

    cout << "Enter height in meters: ";
    cin >> height;

    // Calculate BMI
    double bmi = calculateBMI(weight, height);

    // Display the result
    cout << "BMI: " << bmi << endl;

    // Interpretation of BMI categories
    if (bmi < 18.5) {
        cout << "Underweight" << endl;
    } else if (bmi >= 18.5 && bmi < 24.9) {
        cout << "Normal weight" << endl;
    } else if (bmi >= 25 && bmi < 29.9) {
        cout << "Overweight" << endl;
    } else {
        cout << "Obese" << endl;
    }

    return 0;
}
